cmd/soroban-cli/src/commands/contract/install.rs

```diff
@@ -86,14 +91,35 @@ impl Cmd {
         let (tx_without_preflight, hash) =
             build_install_contract_code_tx(contract.clone(), sequence + 1, self.fee.fee, &key)?;
 
-        client
+        // Currently internal errors are not returned if the contract code is expired
+        if let (
+            TransactionResult {
+                result: TransactionResultResult::TxInternalError,
```

Comment:
does comparison with `TxInternalError` is the best we can do in this case ?
i.e. isn't the comparison here is a bit too broad ?

Reply:
It is, but it is the error returned when the entry is expired.  Would be better to have a specific one, but it's the only one I could match on.  I also tested an install with a random file (see https://github.com/stellar/soroban-tools/issues/787) and it didn't return an error. So I'm not sure what other cases could cause a failure (perhaps a lack of fees?)

Reply:
just a thought for consider, not sure it's feasible, but doing a read attempt at rpc `getLedgerEntries` for the corresponding ContractCode ledger entry first? if absent, same path, but if present, not sure if 'expired' aspect is in the response to check, but then could pro-actively do the restore only if expired rather than a blanket retry due to general error type.